src/folly/FollyKeepaliveTimer.cpp

```diff
@@ -27,8 +27,18 @@ void FollyKeepaliveTimer::schedule() {
   eventBase_.runAfterDelay(
       [this, running]() {
         if (*running) {
-          connection_->sendKeepalive();
-          schedule();
+          if (pending_) {
+            stop();
+
+            connection_->outputFrameOrEnqueue(
```

Comment:
the frame gets written out if we have the allowance, otherwise it is lost and connection closed.